libtransact/src/workload/runner.rs

```diff
@@ -338,7 +340,15 @@ impl WorkerBuilder {
 
                                 // get next target, round robin
                                 next_target = (next_target + 1) % targets.len();
-                                thread::sleep(time_to_wait);
+                                let diff = time::Instant::now() - submission_start;
+                                let submission_time = match submission_avg {
+                                    Some(val) => (diff + val) / 2,
+                                    None => diff,
+                                };
+                                submission_avg = Some(submission_time);
+
+                                thread::sleep(time_to_wait - submission_time);
```

Comment:
I see this throwing the following panic `thread 'Smallbank-Workload-0' panicked at 'overflow when subtracting durations', library/core/src/time.rs:890:31`

Reply:
this happens when the TooManyRequest loop has happened.

Reply:
Yup, should have seen that coming. Changed it to a saturating subtraction so the floor is zero always.